### PR TITLE
added initial code for the correlation function and related tests

### LINKED ISSUE #25: Create test function for calc_corr
Create test function for calc_corr

### LINKED ISSUE #28: Write function body for calc_cor
Write function body for calc_cor

### PR DESCRIPTION
**Pull request by sweber15**
added initial code for the correlation function and related tests
closes #25
closes #28

### CODE REVIEW

File: R/eda.R [hunk 1]
```diff
@@ -79,8 +79,43 @@ describe_cat_var <- function(dataframe, cat_vars) {
...
+  # Test dataframe input to check if dataframe is a dataframe
+  if (!is.data.frame(df))
+    stop("Input 'df' should be a dataframe.")
+
+  # Find numerical columns and remove NA
+  df_num <- df %>%
+    select(num_vars) %>%
+    drop_na()
+
+  # Test colums to check if columns provided are numeric
```
Comment (marvinmin) [inline R/eda.R:148]:
```suggestion
  # Test columns to check if columns provided are numeric
```

File: R/eda.R [hunk 2]
```diff
@@ -79,8 +79,43 @@ describe_cat_var <- function(dataframe, cat_vars) {
...
+  for (i in num_var) {
+    if (typeof(df_num[,i]) == "list") {
+      data_unlisted <- unlist(df_num[i])
+      if (!is.numeric(data_unlisted))
+        stop("Columns do not all contain numeric values.")
+    } else if (!is.numeric(df_num[, i])) {
+      stop("Columns do not all contain numeric values.")
+    }
+  }
```
Comment (marvinmin) [inline R/eda.R]: Does the `num_var` only take list as input? I thought in R a vector of strings is more common.

**Review by marvinmin — CHANGES_REQUESTED** [review]
I think the tests for the error message can be added. It kind of tests all the branches of the `if` statement.

**Comment by sweber15** [comment]
> I think the tests for the error message can be added. It kind of tests all the branches of the `if` statement.

Do you mean something like you did in your testing code?
`  test_that("Corresponding error message should be expected if the dataframe argument is not a dataframe.", {
    expect_error(describe_num_var("abc", num_var), 
    regexp = "The value of the argument 'dataframe' should be of type  'data.frame' or 'tibble'.")
  })`

File: R/eda.R (hunk 2 above)
Comment (sweber15) [reply R/eda.R]: No, it should take a tibble or dataframe. I was having trouble with tibbles, so I put in the condition if it was a list. I could also change it the way you did to make tibbles into dataframes.

**Comment by marvinmin** [comment]
> > I think the tests for the error message can be added. It kind of tests all the branches of the `if` statement.
> 
> Do you mean something like you did in your testing code?
> ` test_that("Corresponding error message should be expected if the dataframe argument is not a dataframe.", { expect_error(describe_num_var("abc", num_var), regexp = "The value of the argument 'dataframe' should be of type 'data.frame' or 'tibble'.") })`

yes, but I don't know if there is a smarter way of doing this.

File: R/eda.R (hunk 2 above)
Comment (marvinmin) [reply R/eda.R]: No, I did't make `tibble` into `data.frame`, but if the input is `data.frame`, I make it into `tibble`. 
Sorry I misread the purpose of this chunk of code before. If you only want to check if only numeric columns are selected, `all(sapply(df_num, is.numeric))` should work.

Comment (sweber15) [reply R/eda.R]: That's much cleaner! Thank you! I will make that change.

Comment (sweber15) [reply R/eda.R]: Added revised code in PR #36

**Comment by sweber15** [comment]
> > > I think the tests for the error message can be added. It kind of tests all the branches of the `if` statement.
> > 
> > 
> > Do you mean something like you did in your testing code?
> > ` test_that("Corresponding error message should be expected if the dataframe argument is not a dataframe.", { expect_error(describe_num_var("abc", num_var), regexp = "The value of the argument 'dataframe' should be of type 'data.frame' or 'tibble'.") })`
> 
> yes, but I don't know if there is a smarter way of doing this.

> I couldn't think of a better way, so I added the tests to my testing document in PR #36

File: R/eda.R (hunk 1 above)
Comment (marvinmin) [reply R/eda.R:148]: There is a small typo

**Review by marvinmin — APPROVED** [review]

**Review by sgauravm — APPROVED** [review]

**Review by v5y8 — APPROVED** [review]
looks good; I like the detailed tests.

**Pull request merged**